daffodil-runtime2/src/main/scala/org/apache/daffodil/runtime2/generators/CodeGeneratorState.scala

```diff
@@ -36,11 +37,32 @@ import scala.collection.mutable
  * Builds up the state of generated code.
  */
 class CodeGeneratorState {
-  private val structs = mutable.Stack[ComplexCGState]()
-  private val prototypes = mutable.ArrayBuffer[String]()
-  private val erds = mutable.ArrayBuffer[String]()
-  private val finalStructs = mutable.ArrayBuffer[String]()
-  private val finalImplementation = mutable.ArrayBuffer[String]()
+  private val elementsAlreadySeen = mutable.Map.empty[String, ElementBase]
```

Comment:
The definition of mutable.Map.empty is:

```scala
def empty[K, V]: Map[K, V] = new HashMap[K, V]
```

The other mutable.\<C\>.empty functions also return new instances.  There is no semantic (only syntactic) difference between empty calls (parentheses are optional), apply calls (parentheses are mandatory), and constructors calls (`new`  is mandatory).  I checked and found all three types of mutable container calls used in Daffodil source code.  Nevertheless, I'll switch back to apply since that idiom is more common and some lines at the end of this same file use apply as well.